### PR TITLE
feat: add permissions for upload and delete separately

### PR DESCRIPTION
**Pull request by gabrielfs7**
feat: add permissions for upload and delete separately

https://oat-sa.atlassian.net/browse/ADF-573

### CODE REVIEW

File: model/mapper/MediaSourcePermissionsMapper.php [hunk 1]
```diff
@@ -22,15 +22,42 @@
...
     /** @var ActionAccessControl */
     private $actionAccessControl;
 
+    public function map(array $data, string $resourceUri): array
+    {
+        $data = parent::map($data, $resourceUri);
+
+        if ($this->hasReadAccessByContext(taoItems_actions_ItemContent::class, 'download')
+            && $this->hasReadAccess($resourceUri)) {
+            $data[parent::DATA_PERMISSIONS][] = self::PERMISSION_DOWNLOAD;
```
Comment (shpran) [inline model/mapper/MediaSourcePermissionsMapper.php:45]: We can use `self::DATA_PERMISSIONS` instead of `parent::DATA_PERMISSIONS`

**Review by shpran — APPROVED** [review]
BE part is good 🙂 
Good job

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: model/mapper/MediaSourcePermissionsMapper.php (hunk 1 above)
Comment (gabrielfs7) [reply model/mapper/MediaSourcePermissionsMapper.php]: True, done!

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**